File: apps/sdkwork-knowledgebase-mini-program/packages/sdkwork-knowledgebase-mp-commons/src/components/screenStates.ts
/** Domain-neutral screen-state vocabulary for mini program pages. */
export type KnowledgebaseMpScreenState = "empty" | "error" | "loading" | "ready" | "unavailable";

export function resolveKnowledgebaseMpScreenStatus(input: {
  error?: string;
  itemCount: number;
  loading: boolean;
}): KnowledgebaseMpScreenState {
  if (input.loading) {
    return "loading";
  }
  if (input.error) {
    return "error";
  }
  return input.itemCount > 0 ? "ready" : "empty";
}

File: apps/sdkwork-knowledgebase-flutter-mobile/packages/sdkwork_knowledgebase_flutter_mobile_commons/lib/src/widgets/screen_states.dart
import 'package:flutter/material.dart';

/// Domain-neutral screen/list state primitives.
///
/// Capability packages map their own data onto these payload-free primitives.
/// Authority: `APP_FLUTTER_UI_SPEC.md`.
enum SdkworkKnowledgebaseScreenStatus { loading, ready, empty, error }

SdkworkKnowledgebaseScreenStatus resolveSdkworkKnowledgebaseScreenStatus(
  int itemCount,
  bool loading,
  String? errorMessage,
) {
  if (loading) {
    return SdkworkKnowledgebaseScreenStatus.loading;
  }
  if (errorMessage != null && errorMessage.isNotEmpty) {
    return SdkworkKnowledgebaseScreenStatus.error;
  }
  return itemCount == 0
      ? SdkworkKnowledgebaseScreenStatus.empty
      : SdkworkKnowledgebaseScreenStatus.ready;
}

/// Payload-free status widget used by capability screens.
class SdkworkKnowledgebaseStatusView extends StatelessWidget {
  const SdkworkKnowledgebaseStatusView({super.key, required this.message});

  final String message;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Text(message, textAlign: TextAlign.center),
      ),
    );
  }
}
